rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -200,7 +201,7 @@ def __init__(
         log.debug(f'Initializing {self.chain_name} inquirer. Nodes to connect {connect_at_start}')
 
         # A cache for the erc20 contract info to not requery same one
-        self.contract_info_cache: dict[ChecksumEvmAddress, dict[str, Any]] = {}
+        self.contract_info_cache: dict[tuple[ChecksumEvmAddress, EvmTokenKind], dict[str, Any]] = {}  # noqa: E501
```

Comment:
That's not what I meant by 2 caches @nebolax 

This approach uses two keys which makes the hashing more complicated than necessary.

It's impossible to have the same address for different stuff in EVM. So the key being only the address is correct. You just need a different map per token kind. So just two different dicts. Or a double dict, but probably 2 different dicts is cleaner?

You would also need to adjust the comment above

Reply:
Yes, makes sense. 2 dicts sounds better to me